File: abc/abc309/b.py
n = int(input())
a = [input() for _ in range(n)]

b = [list(s) for s in a]
for i in range(n - 1):
    b[0][i + 1] = a[0][i]
for i in range(n - 1):
    b[i + 1][-1] = a[i][-1]
for i in range(n - 1):
    b[-1][i] = a[-1][i + 1]
for i in range(n - 1):
    b[i][0] = a[i + 1][0]

for l in b:
    print(''.join(l))

File: abc/abc309/b.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    int n;
    std::cin >> n;
    std::vector<std::string> a(n);
    for (auto &input : a) {
        std::cin >> input;
    }

    auto b = a;
    for (int i = 0; i < n - 1; i++) {
        b[0][i + 1] = a[0][i];
    }
    for (int i = 0; i < n - 1; i++) {
        b[i + 1][n - 1] = a[i][n - 1];
    }
    for (int i = 0; i < n - 1; i++) {
        b[n - 1][n - 2 - i] = a[n - 1][n - 1 - i];
    }
    for (int i = 0; i < n - 1; i++) {
        b[n - 2 - i][0] = a[n - 1 - i][0];
    }

    for (auto &output : b) {
        std::cout << output << std::endl;
    }
    return 0;
}